cli/azd/ci-build.ps1

```diff
@@ -39,43 +39,91 @@ if ($IsWindows) {
     Write-Host "go generate succeeded"
 }
 
+# Run `go help build` to obtain detailed information about `go build` flags.
+$buildFlags = @(
+    # Remove file system paths from the compiled binary
+    "-gcflags=-trimpath",
+    # Remove file system paths from the assembled code
+    "-asmflags=-trimpath",
+    # remove all file system paths from the resulting executable.
+    # Instead of absolute file system paths, the recorded file names
+    # will begin either a module path@version (when using modules),
+    # or a plain import path (when using the standard library, or GOPATH).
+    "-trimpath"
```

Comment:
@wbreza Noticed that `trimpath` was only passed on Windows, but was specified in the PR description of #1950 to be for all platforms. Can you confirm what the desired behavior is? I think "all platforms" is the intended behavior, and made the change as such. Will learn if the build runs successfully in CI